pyscriptjs/tests/integration/test_plugins.py

```diff
@@ -186,3 +186,115 @@ def test_no_plugin_attribute_error(self):
         )
         # EXPECT an error for the missing attribute
         assert error_msg in self.console.error.lines
+
+    def test_fetch_python_plugin(self):
+        """
+        Test that we can fetch a plugin from a remote URL. Note we need to use
+        the 'raw' URL for the plugin, otherwise the request will be rejected
+        by cors policy.
+        """
+        self.pyscript_run(
+            """
+            <py-config>
+                plugins = [
+                    "https://raw.githubusercontent.com/FabioRosado/pyscript-plugins/main/python/hello-world.py"
+                ]
+
+            </py-config>
+            <py-hello-world></py-hello-world>
+            """
+        )
+
+        hello_element = self.page.locator("py-hello-world")
+        assert hello_element.inner_html() == '<div id="hello">Hello World!</div>'
+
+    def test_fetch_js_plugin(self):
+        self.pyscript_run(
+            """
+            <py-config>
+                plugins = [
+                    "https://raw.githubusercontent.com/FabioRosado/pyscript-plugins/main/js/hello-world.js"
+                ]
+            </py-config>
+            """
+        )
+
+        hello_element = self.page.locator("py-hello-world")
+        assert hello_element.inner_html() == "<h1>Hello, world!</h1>"
+
+    def test_fetch_js_plugin_bare(self):
+        self.pyscript_run(
+            """
+            <py-config>
+                plugins = [
+                    "https://raw.githubusercontent.com/FabioRosado/pyscript-plugins/main/js/hello-world-base.js"
+                ]
+            </py-config>
+            """
+        )
+
+        hello_element = self.page.locator("py-hello-world")
+        assert hello_element.inner_html() == "<h1>Hello, world!</h1>"
+
+    def test_fetch_plugin_no_file_extension(self):
+        self.pyscript_run(
+            """
+            <py-config>
+                plugins = [
+                    "http://non-existent.blah/hello-world"
+                ]
+            </py-config>
+            """,
+            wait_for_pyscript=False,
+        )
+
+        expected_msg = (
+            "(PY2000): Unable to load plugin from "
+            "'http://non-existent.blah/hello-world'. Plugins "
+            "need to contain a file extension and be either a "
+            "python or javascript file."
+        )
+
+        assert self.assert_banner_message(expected_msg)
+
+    def test_fetch_js_plugin_non_existent(self):
+        self.pyscript_run(
+            """
+            <py-config>
+                plugins = [
+                    "http://non-existent.blah.com/hello-world.js"
+                ]
+            </py-config>
+            """,
+            wait_for_pyscript=False,
+        )
+
+        expected_msg = (
+            "(PY0001): Fetching from URL "
+            "http://non-existent.blah.com/hello-world.js failed "
```

Comment:
Since 'blah.com' does actually exist (or at least, is registered), how about `example.com/hello-world.js` ?